### ISSUE TITLE
2.3.0: LTO warnings

### ISSUE THREAD

**Issue by kloczek** [Netflix/vmaf#915]
Summary stats of all compile and link time warninfs:
```conseole
[tkloczko@barrel SPECS]$ rpmbuild -ba --with check --with failing_tests vmaf.spec 2>&1 | grep -- \\[-W | sed 's/.*\[//; s/\]//' | sort | uniq -c | sort -nr
     84 -Wincompatible-pointer-types
     64 -Wimplicit-fallthrough=
     58 -Wdiscarded-qualifiers
     40 -Wunused-variable
     20 -Wunused-parameter
     20 -Wlto-type-mismatch
     11 -Wsign-compare
      5 -Wunused-but-set-variable
      3 -Wunknown-pragmas
      2 -Wpointer-arith
      1 -Wtautological-compare
      1 -Wsequence-point
      1 -Wdeprecated-declarations
```
Full log is in attachment.
[vmaf.log](https://github.com/Netflix/vmaf/files/7358344/vmaf.log)

**Comment by kloczek** [comment]
Updates stats for 2.3.1:
```console
[tkloczko@pers-jacek SPECS]$ rpmbuild -ba --with check --with failing_tests vmaf.spec 2>&1 | grep -- \\[-W | sed 's/.*\[//; s/\]//' | sort | uniq -c | sort -nr
     69 -Wincompatible-pointer-types
     64 -Wimplicit-fallthrough=
     37 -Wunused-variable
     20 -Wlto-type-mismatch
     11 -Wsign-compare
      5 -Wunused-but-set-variable
      3 -Wunknown-pragmas
      2 -Wuse-after-free
      1 -Wsequence-point
      1 -Wdeprecated-declarations
```
In this case most imprtand are LTO warnings because it breaks LTO).
It is necessaty to sync ehat is in header files and .c files/
`-Wuse-after-free` should be next.

**Comment by nilfm99** [comment]
I think these were all fixed with #1228. Closing this thread.
